src/rust/engine/fs/store/src/local.rs

```diff
@@ -248,19 +248,76 @@ impl ByteStore {
     dbs?.remove(digest.hash).await
   }
 
+  ///
+  /// Store the given data in a single pass, optionally using the given Digest. Prefer `Self::store`
+  /// for values which should not be pulled into memory, and `Self::store_bytes_batch` when storing
+  /// multiple values at a time.
+  ///
   pub async fn store_bytes(
     &self,
     entry_type: EntryType,
+    digest: Option<Digest>,
     bytes: Bytes,
     initial_lease: bool,
   ) -> Result<Digest, String> {
-    self
-      .store(entry_type, initial_lease, true, move || {
-        Ok(bytes.clone().reader())
-      })
-      .await
+    let dbs = match entry_type {
+      EntryType::Directory => self.inner.directory_dbs.clone(),
+      EntryType::File => self.inner.file_dbs.clone(),
+    };
+    let len = bytes.len();
+    let fingerprint = dbs?
+      .store_bytes(digest.map(|d| d.hash), bytes, initial_lease)
+      .await?;
+
+    Ok(Digest::new(fingerprint, len))
+  }
+
+  ///
+  /// Store the given items in a single pass, optionally using the given Digests. Prefer `Self::store`
+  /// for values which should not be pulled into memory.
+  ///
+  /// See also: `Self::store_bytes`.
+  ///
+  pub async fn store_bytes_batch(
+    &self,
+    entry_type: EntryType,
+    items: Vec<(Option<Digest>, Bytes)>,
+    initial_lease: bool,
+  ) -> Result<Vec<Digest>, String> {
+    let dbs = match entry_type {
```

Comment:
This seems to be a clean helper method candidate.

Reply:
Yea, good call... this was the fourth-or-so copy, and I considered doing it here. Will do in a followup.